ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -76,92 +71,102 @@ Web based editors like Word Online, Excel Online, Google Sheets, etc. may suppor
 
-### 5.1 User permission requirement
+### 4.2 Permissions and Interop - No user permission required 
+ 
+Today browser engines have different approaches to clipboard API permissions. While Chromium has [this permissions model](https://github.com/w3c/clipboard-apis/blob/main/explainer.adoc#clipboard-permissions) for clipboard, [Firefox](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) and [Safari](https://webkit.org/blog/10855/async-clipboard-api/) rely on combination of user activation and user gesture for web pages to access user clipboard contents. This strict requirement is present because a user's clipboard contents are highly sensitive and can contain private data like passwords, security tokens. However, we will see that core scenarios involving clipboardchange event does not require complete user clipboard contents and hence we propose a more lenient permissions approach, that works in all browsers, for this new API. 
+ 
+**Permission prompt in Chromium**
+![](img/chrome-permission-prompt.png)
 
-#### 5.1.1 Approach 1 (Preferred) - clipboard-read permission required to listen to clipboardchange event
-Since the clipboard contains privacy-sensitive data, we should protect access to the clipboard change event using a user permission - clipboard-read. The web author should ensure that the site has the permission before it starts listening to this event otherwise the provided event handler won't be invoked whenever the clipboard changes. To check if the current user has clipboard-read permissions for the site, the [query](https://www.w3.org/TR/permissions/#query-method) method of the [Permissions API](https://www.w3.org/TR/permissions/#permissions-api) can be used. We should consider logging a warning message if the web author starts listening to clipboardchange without acquiring the permissions since web developers might miss integrating the permissions flow into their user experience.
+**Permission prompt in Firefox**
+![](img/firefox-paste-button.png)
+ 
+With ClipboardChange event API, event would be fired to indicate there was a change to clipboard. Without any additional data or information provided with this event, there shouldn't be any privacy concern as there is no data exposed. As no data is exposed, no permissions are needed. Now let’s consider the use cases outlined in the explainer, especially the dynamic contextual menus and buttons appearing or not appearing with different MIME types available on clipboard([Scenario 2.2](#21-scenario-show-available-paste-formats-in-web-based-editors)). For this just firing the clipboard change event is not sufficient. Developers will need to know the available MIME types on clipboard to provide such capabilities on Web Platform. And a developer can know the available types if the browser provides Clipboard MIME types as part of payload on the ClipboardChange event.
```

Comment:
My general feedback here is that I think most of this can stated much more succinctly. Ex:

"The ClipboardChange event object will have a `types` member that lists all the available native formats available on the clipboard. Custom formats will not be included in this list to minimize fingerprinting risk."

You can then mention in 1-2 sentences on how the `types` can be used.